Tests/MacroToolkitTests/DeclGroupTests.swift

```diff
@@ -0,0 +1,161 @@
+        let decl: DeclSyntax = """
+        extension TestStruct { func testMethod() {} }
+        """
+        let extensionDecl = decl.as(ExtensionDeclSyntax.self)!
+        let testExtension = Extension(extensionDecl)
+        
+        XCTAssertEqual(testExtension.identifier, "TestStruct")
+        XCTAssertEqual(testExtension.members.count, 1)
+    }
+    
+    func testDeclGroupInitialization() throws {
+        let structDecl: DeclSyntax = """
+        struct TestStruct { var value: Int }
+        """
+        let structSyntax = structDecl.as(StructDeclSyntax.self)!
+        let structDeclGroup = DeclGroup(structSyntax)
+        
+        switch structDeclGroup {
+        case .struct(let testStruct):
+            XCTAssertEqual(testStruct.identifier, "TestStruct")
+            XCTAssertEqual(testStruct.members.count, 1)
+            XCTAssertEqual(testStruct.properties.count, 1)
+        default:
+            XCTFail("Expected .struct case")
+        }
+        
+        let enumDecl: DeclSyntax = """
+        enum TestEnum { case caseOne, caseTwo }
+        """
+        let enumSyntax = enumDecl.as(EnumDeclSyntax.self)!
+        let enumDeclGroup = DeclGroup(enumSyntax)
+        
+        switch enumDeclGroup {
+        case .enum(let testEnum):
+            XCTAssertEqual(testEnum.identifier, "TestEnum")
+            XCTAssertEqual(testEnum.members.count, 1)
+            XCTAssertEqual(testEnum.cases.count, 2)
+        default:
+            XCTFail("Expected .enum case")
+        }
+        
+        let classDecl: DeclSyntax = """
+        class TestClass { var value: Int }
+        """
+        let classSyntax = classDecl.as(ClassDeclSyntax.self)!
+        let classDeclGroup = DeclGroup(classSyntax)
+        
+        switch classDeclGroup {
+        case .class(let testClass):
+            XCTAssertEqual(testClass.identifier, "TestClass")
+            XCTAssertEqual(testClass.members.count, 1)
+            XCTAssertEqual(testClass.properties.count, 1)
+        default:
+            XCTFail("Expected .class case")
+        }
+        
+        let actorDecl: DeclSyntax = """
+        actor TestActor { var value: Int }
+        """
+        let actorSyntax = actorDecl.as(ActorDeclSyntax.self)!
+        let actorDeclGroup = DeclGroup(actorSyntax)
+        
+        switch actorDeclGroup {
+        case .actor(let testActor):
+            XCTAssertEqual(testActor.identifier, "TestActor")
+            XCTAssertEqual(testActor.members.count, 1)
+            XCTAssertEqual(testActor.properties.count, 1)
+        default:
+            XCTFail("Expected .actor case")
+        }
+        
+        let extensionDecl: DeclSyntax = """
+        extension TestStruct { func testMethod() {} }
+        """
+        let extensionSyntax = extensionDecl.as(ExtensionDeclSyntax.self)!
+        let extensionDeclGroup = DeclGroup(extensionSyntax)
+        
+        switch extensionDeclGroup {
+        case .extension(let testExtension):
```

Comment:
Indent switch cases one level (just cause that's the formatting I'm using for this repo)